src/btree/bt_random.c

```diff
@@ -358,13 +389,14 @@ __wt_random_descent(WT_SESSION_IMPL *session, WT_REF **refp, uint32_t flags, WT_
     WT_PAGE_INDEX *pindex;
     WT_REF *current, *descent;
     uint32_t i, entries, retry;
```

Comment:
One additional conflict here: On develop `retry` is of type `int`, not `uint32_t`, due to WT-12708. 
I've kept the 7.0 implementation that uses `uint32_t`